kotlin-asyncapi-context/src/main/kotlin/org/openfolder/kotlinasyncapi/context/annotation/AnnotationProvider.kt

```diff
@@ -67,19 +68,31 @@ class AnnotationProvider(
 
         annotatedClasses
             .flatMap { clazz ->
-                listOfNotNull(
+                val classAnnotations = listOfNotNull(
                     clazz.findAnnotation<Message>()?.let { clazz to it },
                     clazz.findAnnotation<Schema>()?.let { clazz to it },
                     clazz.findAnnotation<Channel>()?.let { clazz to it }
                 )
+
+                val functionAnnotations = clazz.functions.flatMap { function ->
```

Comment:
This will only look at functions of classes that are annotated with a `AsyncApiAnnotation` (see above). We could add a new generic annotation `AsyncApiComponent`. Then you would still need to mark the Spring configuration class with `@AsyncApiComponent`. wdyt?

Reply:
Ah I see what you're saying. My vote would be for AsyncApiDocumentation over *Component, the only reason I'm being particular is that I don't want folks to see it and think it's a Spring managed bean. It's not a strong opinion though so if your preference is Component, I can live with it.

Reply:
This has been updated. Please LMK what you think.